// src/app/app.component.ts
import { Component, OnInit, Testability } from '@angular/core';
import { analyzeAndValidateNgModules } from '../../node_modules/@angular/compiler';
import { Alert } from '../../node_modules/@types/selenium-webdriver';
import { trigger, style, transition, animate, keyframes, query, stagger, animation} from '@angular/animations';
import { TestingCompiler } from '../../node_modules/@angular/core/testing/src/test_compiler';
import { HttpClient } from '@angular/common/http';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
 /* animations: [ 
    
   
    trigger('goals', [
      //trigger content
      transition('* => *', [
        query(':enter', style({ opacity: 0}), {optional: true}),
        query(':enter', stagger('300ms', [
          animate('0.6 ease-in', keyframes([
            style({opacity: 0, transfrom: 'translateY(-75%)', offset: 0 }),
            style({opacity: 0.5, transfrom: 'translateY(35px)', offset: 0.3 }),
            style({opacity: 1, transfrom: 'translateY(0)', offset: 1 }),
          ]))]), {optional: true})
      ])

  ]),//end of trigger

  ]*/
})
export class AppComponent{

constructor(private httpClient:HttpClient){


}
name:String = '';
NoFrillsprice:number;
NoFrillsamount:number;
NoFrillsname:string;
Loblawsprice:number;
Loblawsamount:number;
Loblawsname:string;
found:boolean;

onNameKeyUp(event: any){
  
  this.name = event.target.value;
  this.found = false;
  
}

getGroceries(){
this.httpClient.get('https://my-json-server.typicode.com/d3c0r33/devorii/grocery/?product=' + this.name)
.subscribe(
(data:any[]) => { 
  console.log(data);

  if(data.length){
    this.NoFrillsname = data[0].NoFrillsname;
    this.NoFrillsprice = data[0].price;
    this.NoFrillsamount = data[0].amount;
    this.Loblawsname = data[1].Loblawsname;
    this.Loblawsprice = data[1].price;
    this.Loblawsamount = data[1].amount;
    this.found = true;

  } 
  
} 
)
}

 

title = "Personal Dashboard" 
submit = "Let's Budget"
TodayGoals = " Today goals..."
dummytext = "Do yoga this morning :)"
nest = null;
task = [];
sumBudget = 0;
titleTwo = "Finances - Monthly Budget";
titleThree = "Search Groceries";

home: number = 0;
hydro: number = 0;
food: number = 0;
kids: number = 0;
laundry: number = 0;
car: number = 0;
investment: number = 0;
debt: number = 0;
creditCard: number = 0;
extras: number = 0;
savings: number = 0;
Income: number = 0;
income = []
saving = [];
remainingCash = [];
total = [];





 addItem(){ 
  this.task.push(this.dummytext); 
  this.dummytext = "";
  if(this.task.length == 5){ 
    alert();   
  }
}

calcuFinances(){ 

  if(this.home > 0){
   this.total.push(this.home + this.hydro + this.food + this.kids + this.laundry +
   this.car + this.investment + this.debt + this.creditCard + this.extras);

   this.saving.push(this.savings);

   this.income.push(this.Income - (this.Income * 15 / 100) );

   this.remainingCash.push(this.Income - (this.savings + this.home + this.hydro + this.food + this.kids + this.laundry +
    this.car + this.investment + this.debt + this.creditCard + this.extras))



   alert("15% tax was deducted due to the candian tax laws for more information visit www.google.com");
  }else if (this.home == 0){
    alert("Home field must be completed for this form, please refresh the page.");
  }

}

}
